packages/vc-verification/src/resolver/Ipfs-credential-resolver.ts

```diff
@@ -81,17 +86,22 @@ export class IpfsCredentialResolver implements CredentialResolver {
    *  provider,
    *  registrySettings,
    *  didStore );
-   * const credential = credentialResolver.getVerifiableCredential('did:ethr:1234', 'sampleRole');
+   * const credential = credentialResolver.getVerifiableCredential('did:ethr:1234', 'sampleRole', roleCredentialCache);
    * ```
    *
    * @param did subject DID for which the credential needs to be fetched
    * @param namespace role for which the credential needs to be fetched
+   * @param roleCredentialCache Cache to store role credentials
    * @returns
    */
-  async getVerifiableCredential(did: string, namespace: string) {
+  async getVerifiableCredential(
+    did: string,
+    namespace: string,
+    roleCredentialCache: IRoleCredentialCache
+  ) {
     const credentials = await this.credentialsOf(did);
     credentials.forEach((credential) =>
-      this._roleCredentialCache?.setRoleCredential(
+      roleCredentialCache.setRoleCredential(
```

Comment:
if roleCredentialCache is optional param, an additional check for it is necessary here